drivers/ethernet/eth_stm32_hal.c

```diff
@@ -98,43 +203,78 @@ static int eth_tx(struct device *dev, struct net_pkt *pkt)
 	ETH_HandleTypeDef *heth;
 	uint8_t *dma_buffer;
 	int res;
-	uint16_t total_len;
+	uint32_t u32_total_len;
```

Comment:
What is the purpose of this change? If you are worried about the size of the var, just change the type of total_len to size_t.

Reply:
> What is the purpose of this change? If you are worried about the size of the var, just change the type of total_len to size_t.

changed to `size_t`

Reply:
changed to `size_t`